tfjs-backend-webgpu/src/shader_preprocessor_wgsl.ts

```diff
@@ -0,0 +1,604 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {backend_util, DataType, util} from '@tensorflow/tfjs-core';
+import {symbolicallyComputeStrides} from './shader_util';
+
+export function getCoordsDataTypeWgsl(rank: number): string {
```

Comment:
Since you are using `uint` instead of `int` in wgsl, should you change to use `uniformDataView.setUInt32` for wgsl in https://github.com/tensorflow/tfjs/blob/master/tfjs-backend-webgpu/src/backend_webgpu.ts#L542?

Reply:
Yes, updated